### PR TITLE
Redirect to the ?next= url after authenticating

### PR DESCRIPTION
**Pull request by lgarvey**
This PR fixes a bug that prevented the user from being redirected to the original page after authenticating.

### CODE REVIEW

**Comment by stuaxo** [comment]
I guess there's no chance the redirect url could have parameters already ?

**Comment by lgarvey** [comment]
If the current URL has a ?next query string I don't think we'd want to
redirect to that instead.  However, we'd probably want to retain the query
string parameters on the current URL. What do you think?

**Comment by stuaxo** [comment]
Yep that expresses it better. 

I really only scan read this so there's a good chance I got the wrong end of the stick :)

File: authbroker_client/views.py [hunk 1]
```diff
@@ -2,22 +2,44 @@
...
 except ImportError:
     from sentry_sdk import capture_exception
 
 
+def get_next_url(request):
+    next_url = request.GET.get(
+        REDIRECT_FIELD_NAME,
+        request.session.get(REDIRECT_SESSION_FIELD_NAME)
+    )
+    if next_url and is_safe_url(next_url, allowed_hosts=settings.ALLOWED_HOSTS):
```
Comment (mcbhenwood) [inline authbroker_client/views.py:32]: Nice - `and is_safe_url()`

Comment (mcbhenwood) [reply authbroker_client/views.py:32]: saying that tho, do you maybe want to add a test to check that an 'unsafe' URL is _not_ returned as `next`?

**Review by mcbhenwood — CHANGES_REQUESTED** [review]
Maybe good to add the extra test I mention?

File: authbroker_client/views.py (hunk 1 above)
Comment (stuaxo) [reply authbroker_client/views.py:32]: Always the key to testing ... put on evil hat and think of all the ways to trip up the code :+1: 

It looks like `is_safe_url` is getting deprecated in favour of `url_has_allowed_host_and_scheme`.

Comment (lgarvey) [reply authbroker_client/views.py]: Yep, I've added to a couple of tests to verify it only handles safe urls

File: tests/test_views.py [hunk 2]
```diff
@@ -27,6 +27,16 @@ def test_auth_view_retains_next_url(client):
     assert client.session[REDIRECT_SESSION_FIELD_NAME] == '/go-here-after-logging-in/'
 
 
+@pytest.mark.django_db
+def test_auth_view_retains_unsafe_next_url(client):
+    url = reverse('authbroker:login') + '?next=https://danger.com'
+    response = client.get(url)
+    assert response.status_code == 302
+    assert AUTHORISATION_URL in response.url
+    assert 'code=someCode' not in response.url
```
Comment (mcbhenwood) [inline tests/test_views.py:36]: maybe you meant `assert 'next=https://danger.com' not in response.url` ?

Comment (lgarvey) [reply tests/test_views.py]: Nah, that was just an artifact from the test I copied. I've removed it.

**Review by mcbhenwood — APPROVED** [review]

**Pull request merged**